File: LeetCodeJava/solution/src/main/java/com/cyanflxy/leetcode/_5/_507.java
package com.cyanflxy.leetcode._5;

/**
 * @author cyanflxy
 * @since 2020/9/8
 */
public class _507 {
    public boolean checkPerfectNumber(int num) {
        if (num <= 3) {
            return false;
        }
        int result = 1;
        int i = 2;

        for (; i * i < num; i++) {
            if (num % i == 0) {
                result += i;
                result += num / i;
            }
        }

        if (i * i == num) {
            result += i;
        }
        return result == num;
    }
}

File: LeetCodeJava/src/com/cyanflxy/leetcode/_5/_507.kt
package com.cyanflxy.leetcode._5

fun convertToBase7(num: Int): String {
    if (num == 0) {
        return "0"
    }

    val sign = num < 0
    var v = if (num >= 0) num else -num
    val sb = StringBuilder()

    while (v > 0) {
        val i = v % 7
        sb.append(i)
        v /= 7
    }

    if (sign) {
        sb.append('-')
    }
    return sb.reverse().toString()
}